Loop/View Controllers/StatusTableViewController.swift

```diff
@@ -98,7 +98,6 @@ final class StatusTableViewController: LoopChartsTableViewController {
                     if let preferredGlucoseUnit = self?.deviceManager.glucoseStore.preferredUnit {
                         self?.preferredGlucoseUnit = preferredGlucoseUnit
                         self?.unitPreferencesDidChange(to: preferredGlucoseUnit)
-                        self?.notifyObserversOfPreferredGlucoseUnitChange(to: preferredGlucoseUnit)
```

Comment:
If LoopDataManager is the object responsible for managing glucose unit subscriptions, it should be the one subscribing to the HealthStorePreferredGlucoseUnitDidChange message, and StatusTableViewController should be just another subscriber.

Reply:
Also, when subscribing to HealthStorePreferredGlucoseUnitDidChange, you can specify a queue, so you can specify .main there, and avoid the async dispatch.

Reply:
yes, `HealthStorePreferredGlucoseUnitDidChange` is reported on `.main`.

Sure, I'll update as suggested and let you know when it is ready to review.